src/interceptors/WebSocket/WebSocketClient.ts

```diff
@@ -0,0 +1,73 @@
+/**
+ * WebSocket client class.
+ * This represents an incoming WebSocket client connection.
+ * @note Keep this class implementation-agnostic because it's
+ * meant to be used over any WebSocket implementation
+ * (not all of them follow the one from WHATWG).
+ */
+import type {
+  WebSocketSendData,
+  WebSocketTransport,
+} from './WebSocketTransport'
+import { WebSocketMessageListener } from './implementations/WebSocketClass/WebSocketClassInterceptor'
+import { bindEvent } from './utils/bindEvent'
+
+const kEmitter = Symbol('kEmitter')
+
+/**
+ * The WebSocket client instance represents an incoming
+ * client connection. The user can control the connection,
+ * send and receive events.
+ */
+export class WebSocketClient {
```

Comment:
This is a WebSocket client connection class. It represents a single WebSocket client connected to the "server" (the interceptor). 

This API is implementation-agnostic and provides the public methods to interact with the intercepted WebSocket client.